query.go

```diff
@@ -288,11 +288,12 @@ func newVideoSourceFromDriver(
 		return nil, errors.New("driver not a driver.VideoRecorder")
 	}
 
-	if driverStatus := videoDriver.Status(); driverStatus != driver.StateClosed {
+	if ok, err := driver.IsAvailable(videoDriver); !errors.Is(err, availability.ErrUnimplemented) && !ok {
+		return nil, errors.Wrap(err, "video driver not available")
```

Comment:
Updates how we check for state. A workaround for device driver states being set to `StateClosed` on re-discovery.